### PR TITLE
etcd: Updated to v3.5.4

### LINKED ISSUE #109707: etcd: Update to v3.5.4
Please publish 3.5.4-0 image and update to [etcd v3.5.4](https://github.com/etcd-io/etcd/releases/tag/v3.5.4) to benefit from fix https://github.com/etcd-io/etcd/pull/13950 for https://github.com/etcd-io/etcd/issues/13948, while still keeping v3.5.3 critical correctness fixes (https://github.com/etcd-io/etcd/issues/13766, https://github.com/etcd-io/etcd/issues/13514).

For recent PRs for update to etcd v3.5.3 see https://github.com/kubernetes/kubernetes/pull/109470 and https://github.com/kubernetes/kubernetes/pull/109471

### PR DESCRIPTION
**Pull request by mk46**
#### What type of PR is this?

Add one of the following kinds:
/kind bug
/kind cleanup

#### What this PR does / why we need it:
Update etcd to [v3.5.4](https://github.com/etcd-io/etcd/releases/tag/v3.5.4).
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/109707

#### Special notes for your reviewer:
/hold dependency on https://github.com/kubernetes/kubernetes/pull/110031 
#### Does this PR introduce a user-facing change?

```release-note
- etcd: Update to v3.5.4
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
Note, I have two backports of .3 in the queue but these are not apparoved yet.

https://github.com/kubernetes/kubernetes/pull/109533
https://github.com/kubernetes/kubernetes/pull/109532

Overall +1 to update etcd in k8s more frequently.

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -499,8 +499,9 @@ var (
 		20: "3.4.13-0",
 		21: "3.4.13-0",
 		22: "3.5.3-0",
-		23: "3.5.3-0",
-		24: "3.5.3-0",
+		23: "3.5.4-0",
+		24: "3.5.4-0",
```
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go:503]: Please change all 3.5.x versions in kubeadm to 3.5.4

Comment (mk46) [reply cmd/kubeadm/app/constants/constants.go:503]: Done.

**Comment by mk46** [comment]
/hold cancel
@dims fyi

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**